asb-native/src/main/java/org/ballerinalang/asb/connection/ConUtils.java

```diff
@@ -0,0 +1,278 @@
+        if (map.containsKey("replyTo")) {
+            replyTo = (String) map.get("replyTo");
+        }
+        if (map.containsKey("label")) {
+            label = (String) map.get("label");
+        }
+        if (map.containsKey("sessionId")) {
+            sessionId = (String) map.get("sessionId");
+        }
+        if (map.containsKey("correlationId")) {
+            correlationId = (String) map.get("correlationId");
+        }
+        if (map.containsKey("timeToLive")) {
+            timeToLive = Integer.parseInt(map.get("timeToLive"));
+        }
+
+        try {
+            // Send messages to queue
+            LOG.info("\tSending messages to ...\n" + sender.getEntityPath());
+            IMessage message = new Message();
+            message.setMessageId(messageId);
+            message.setTimeToLive(Duration.ofMinutes(timeToLive));
+            byte[] byteArray = content.getBytes();
+            message.setBody(byteArray);
+            message.setContentType(contentType);
+            message.setMessageId(messageId);
+            message.setTo(to);
+            message.setReplyTo(replyTo);
+            message.setLabel(label);
+            message.setSessionId(sessionId);
+            message.setCorrelationId(correlationId);
+            Map<String,String> propertiesMap = toStringMap(properties);
+            message.setProperties(propertiesMap);
+
+            sender.send(message);
+            LOG.info("\t=> Sent a message with messageId \n" + message.getMessageId());
+        } catch (Exception e) {
+            throw AsbUtils.returnErrorValue(e.getMessage());
+        }
+    }
+
+    /**
+     * Receive Message with configurable parameters as Map when Receiver Connection is given as a parameter and
+     * message content as a byte array and return Message object.
+     *
+     * @param receiver Output Receiver connection.
+     * @return Message Object of the received message.
+     */
+    public static Object receiveMessage(IMessageReceiver receiver) throws Exception {
+        try {
+            LOG.info("\n\tWaiting up to 5 seconds for messages from ...\n" + receiver.getEntityPath());
+
+            IMessage receivedMessage = receiver.receive(Duration.ofSeconds(5));
+
+            if (receivedMessage == null) {
+                return null;
+            }
+            LOG.info("\t<= Received a message with messageId \n" + receivedMessage.getMessageId());
+            LOG.info("\t<= Received a message with messageBody \n" +
+                    new String(receivedMessage.getBody(), UTF_8));
+            receiver.complete(receivedMessage.getLockToken());
+
+            LOG.info("\tDone receiving messages from \n" + receiver.getEntityPath());
+
+            BObject messageBObject = BValueCreator.createObjectValue(AsbConstants.PACKAGE_ID_ASB,
+                    AsbConstants.MESSAGE_OBJECT);
+            messageBObject.set(AsbConstants.MESSAGE_CONTENT, BValueCreator.createArrayValue(receivedMessage.getBody()));
+
```

Comment:
Can remove unwanted newlines. Add newline only if they are necessary.

Reply:
resolved. Thank you